cassovary-core/src/main/scala/com/twitter/cassovary/util/io/AdjacencyListGraphReader.scala

```diff
@@ -80,7 +83,10 @@ class AdjacencyListGraphReader[T] (
     override def iterator = new Iterator[NodeIdEdgesMaxId] {
 
       var lastLineParsed = 0
-      private val src = Source.fromFile(filename)
+      private val src = if (!isGzip)
```

Comment:
this is replicated code from FileReader. It was already there before your changes but now this extra code is making it worse. Can we instantiate a FileReader just like we instantiate TwoTsFileReader in ListOfEdgesGraphReader?